cocotb/types/array.py

```diff
@@ -285,42 +329,20 @@ def __setitem__(self, item, value):
                 "indexes must be ints or slices, not {}".format(type(item).__name__)
             )
 
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, type(self)):
+            if len(self) != len(other):
+                return False
+            return all(a == b for a, b in zip(self, other))
+        return NotImplemented
+
     def __repr__(self) -> str:
         return "{}({!r}, {!r})".format(type(self).__name__, self._value, self._range)
 
-    def __concat__(self, other: "Array") -> "Array":
-        if not isinstance(other, type(self)):
-            return NotImplemented
-        return type(self)(self._value + other._value)
-
-    def __rconcat__(self, other: "Array") -> "Array":
-        if not isinstance(other, type(self)):
-            return NotImplemented
-        return type(self)(other._value + self._value)
-
-    def index(
-        self, value: Any, start: Optional[int] = None, stop: Optional[int] = None
-    ) -> int:
-        """
-        Return index of first occurrence of *value*.
-
-        Raises :exc:`ValueError` if the value is not found.
-        Search only within *start* and *stop* if given.
-        """
-        if start is not None:
-            start = self._translate_index(start)
-        else:
-            start = 0
-        if stop is not None:
-            stop = self._translate_index(stop)
-        else:
-            stop = maxsize  # same default value used by Python lists
-        idx = self._value.index(value, start, stop)
-        return self._range[idx]
-
-    def count(self, value: Any) -> int:
-        """Return number of occurrences of *value*."""
-        return self._value.count(value)
+    def __matmul__(self, other: ArrayABC[S]) -> "Array[typing.Union[T, S]]":
```

Comment:
There is no doc about concatenating Arrays anymore now.

Reply:
I removed that change.